compose/legacy.py

```diff
@@ -86,34 +126,66 @@ def migrate_project_to_labels(project):
 def get_legacy_containers(
         client,
         project,
-        services,
-        stopped=False,
-        one_off=False):
+        services):
+
+    containers = client.containers(all=True)
 
-    containers = client.containers(all=stopped)
+    legacy_containers = []
+    legacy_one_off_containers = []
 
     for service in services:
         for container in containers:
+            if LABEL_VERSION in container['Labels']:
+                continue
+
             name = get_container_name(container)
-            if has_container(project, service, name, one_off=one_off):
-                yield Container.from_ps(client, container)
+            is_ours, is_one_off = check_name(project, service, name)
+
+            if not is_ours:
+                continue
 
+            container = Container.from_ps(client, container)
+
+            if is_one_off:
+                legacy_one_off_containers.append(container)
+            else:
+                legacy_containers.append(container)
+
+    return (legacy_containers, legacy_one_off_containers)
```

Comment:
I think this implementation could be a little simpler.  Instead of always returning both, it could take a `one_off=False` kwarg (like a bunch of functions do currently).

If it's True return only the one_off containers, of it's False only the non-one_off containers.

That removes the need for the tuple return in `check_name`, `validate_name`, and this function.

I removes the unused return value from `migrate_project_to_labels`, and the logic in `check_for_legacy_containers` will be about the same.